client/my-sites/plans-features-main/components/plan-upsell-modal/paid-plan-is-required-dialog.tsx

```diff
@@ -1,74 +1,82 @@
 import { LoadingPlaceholder } from '@automattic/components';
+import { FREE_THEME } from '@automattic/design-picker';
 import { PlanButton } from '@automattic/plans-grid-next';
 import { useEffect, useState } from '@wordpress/element';
 import { useTranslate } from 'i18n-calypso';
 import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
-import {
-	ButtonContainer,
-	DialogContainer,
-	Heading,
-	SubHeading,
-	Row,
-	RowWithBorder,
-	DomainName,
-} from './components';
+import { getHidePlanPropsBasedOnThemeType } from '../utils/utils';
+import { ButtonContainer, DialogContainer, Heading, Row } from './components';
+import { PlanName, PlanDescription } from './components/plan-item';
 import SuggestedPlanSection from './components/suggested-plan-section';
 import { DomainPlanDialogProps, MODAL_VIEW_EVENT_NAME } from '.';
 
-export default function PaidPlanIsRequiredDialog( {
+export const PaidPlanIsRequiredDialog = ( {
 	paidDomainName,
 	generatedWPComSubdomain,
+	selectedThemeType,
 	onFreePlanSelected,
 	onPlanSelected,
-}: DomainPlanDialogProps ) {
+}: DomainPlanDialogProps ) => {
 	const translate = useTranslate();
 	const [ isBusy, setIsBusy ] = useState( false );
+	const isPaidTheme = selectedThemeType && selectedThemeType !== FREE_THEME;
+	const hidePlanProps = getHidePlanPropsBasedOnThemeType( selectedThemeType );
```

Comment:
```suggestion
	const hidePlanProps = getHidePlanPropsBasedOnThemeType( selectedThemeType || '' );
```
(Seeing a type error)